util/logger.go

```diff
@@ -1,55 +1,277 @@
 }
 
-func (l *Logger) Info(msg string) {
-	l.Log(INFO, msg)
+func (l *LoggerService) FatalCtx(req *http.Request, msg string, err error) {
+	event := l.Logger.WithLevel(zerolog.FatalLevel).Err(err)
+	l.enrich(req, event).Msg(msg)
 }
 
-func (l *Logger) Warn(msg string) {
-	l.Log(WARN, msg)
+func (l *LoggerService) PanicCtx(req *http.Request, msg string, r any, trace string) {
+	event := l.Logger.WithLevel(zerolog.InfoLevel).
+		Str("panic_value", fmt.Sprintf("%v", r)).
+		Str("trace", trace)
+	l.enrich(req, event).Msg(msg)
 }
 
-func (l *Logger) Error(msg string) {
-	l.Log(ERROR, msg)
+func (l *LoggerService) SuccessCtx(req *http.Request) {
+	event := l.Logger.WithLevel(zerolog.InfoLevel)
+	l.enrich(req, event).Msg("request successful")
 }
+
+func (l *LoggerService) Debug(msg string) {
+	if l.Env == "PRODUCTION" {
+		return
+	}
+	l.Logger.WithLevel(zerolog.DebugLevel).Msg(msg)
+}
+
+func (l *LoggerService) Info(msg string) {
+	l.Logger.WithLevel(zerolog.InfoLevel).Msg(msg)
+}
+
+func (l *LoggerService) Warn(msg string) {
+	l.Logger.WithLevel(zerolog.InfoLevel).Msg(msg)
+}
+
+func (l *LoggerService) Error(msg string, err error) {
+	l.Logger.WithLevel(zerolog.InfoLevel).Err(err).Msg(msg)
+}
+
+func (l *LoggerService) Fatal(msg string, err error) {
+	l.Logger.WithLevel(zerolog.FatalLevel).Err(err).Msg(msg)
+}
+
+// loggingResponseWriter is a wrapper around http.ResponseWriter to capture status code and response size.
+type loggingResponseWriter struct {
+	http.ResponseWriter
+	statusCode int
+	size       int
+}
+
+func (lrw *loggingResponseWriter) WriteHeader(code int) {
+	lrw.statusCode = code
+	lrw.ResponseWriter.WriteHeader(code)
+}
+
+func (lrw *loggingResponseWriter) Write(b []byte) (int, error) {
+	if lrw.statusCode == 0 {
+		lrw.statusCode = http.StatusOK
+	}
+	size, err := lrw.ResponseWriter.Write(b)
+	lrw.size += size
+	return size, err
+}
+
+func (l *LoggerService) LogMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+		lrw := &loggingResponseWriter{ResponseWriter: w, statusCode: 0, size: 0}
+		next.ServeHTTP(lrw, r)
+
+		// Get client IP from headers or remote address
+		clientIP := r.Header.Get("X-Forwarded-For")
+		if clientIP == "" {
+			clientIP = r.Header.Get("X-Real-IP")
+		}
+		if clientIP == "" {
+			clientIP, _, _ = net.SplitHostPort(r.RemoteAddr)
+		}
+		// In case of multiple IPs in X-Forwarded-For, take the first one
+		if strings.Contains(clientIP, ",") {
+			clientIP = strings.TrimSpace(strings.Split(clientIP, ",")[0])
+		}
+
+		event := l.Logger.WithLevel(zerolog.InfoLevel).
+			Str("route", r.URL.Path).
+			Str("method", r.Method).
+			Int("status", lrw.statusCode).
+			Int("response-size", lrw.size).
+			Dur("duration", time.Since(start)).
+			Interface("query-params", r.URL.Query()).
+			Str("ip", clientIP).
+			Str("user-agent", r.Header.Get("User-Agent"))
+		
+		event.Send()
+	})
+}
+
+var LogVar, err = InitLogger("PRODUCTION")
```

Comment:
Rename to `Logger` perhaps? or `SharedLogger`. `LogVar` sounds odd and confusing of the purpose it serves.